contracts/README.md

```diff
@@ -2,57 +2,54 @@
 Interface that defines the events bridge contract implementations must emit. Also defines the message types and formats of messages between all implementations.
 
-### `IERC20Bridge` and `INativeTokenBridge`
+### `IERC20TokenBridge` and `INativeTokenBridge`
 Interfaces that define the external functions for interacting with bridge contract implementations of each type. ERC20 and native token bridge interfaces vary from each other in that the native token bridge functions are `payable` and do not take an explicit amount parameter (it is implied by `msg.value`), while the ERC20 token bridge functions are not `payable` and require the explicit amount parameter. Otherwise, they include the same functions.
 
-### `TeleporterTokenSource`
-An abstract implementation of `ITeleporterTokenBridge` for a bridge contract on the home chain with the asset to be bridged. Handles locking tokens to be sent to destination chains, as well as receiving bridge messages to either redeem tokens it holds as collateral (i.e unlock), or route them to another chain (i.e. "multi-hop"). In the case of a multi-hop transfer, the `TeleporterTokenSource` already has the collateral locked from when the tokens were originally bridged to the first destination chain, so it simply updates the accounting of the transferred balances to each respective destination. Destination contracts must first be registered with a `TeleporterTokenSource` instance before the source contract will allow for sending tokens to them. This is to prevent tokens from being bridged to invalid destination addresses. Anyone is able to deploy and register destination contracts, which may have been modified from this repository. It is the responsibility of the users of the source contract to independently evaluate each destination for their security and correctness.
+### `TokenHub`
+An abstract implementation of `ITokenBridge` for a bridge contract on the "hub" chain with the asset to be bridged. Each `TokenHub` instance supports bridging exactly one token type (ERC20 or native) on its chain to arbitrarily many "spoke" chains. It handles locking tokens to be sent to spoke chains, as well as receiving bridge messages to either redeem tokens it holds as collateral (i.e unlock), or route them to other spoke chains (i.e. "multi-hop"). In the case of a multi-hop transfer, the `TokenHub` already has the collateral locked from when the tokens were originally bridged to the first destination chain, so it simply updates the accounting of the transferred balances to each respective spoke. Spoke contracts must first be registered with a `TokenHub` instance before the hub contract will allow for sending tokens to them. This is to prevent tokens from being bridged to invalid spoke addresses. Anyone is able to deploy and register spoke contracts, which may have been modified from this repository. It is the responsibility of the users of the hub contract to independently evaluate each spoke for its security and correctness.
```

Comment:
```suggestion
An abstract implementation of `ITokenBridge` for a bridge contract on the "hub" chain with the asset to be bridged. Each `TokenHub` instance supports bridging exactly one token type (ERC20 or native) on its chain to arbitrarily many "spoke" chains. It handles locking tokens to be sent to spoke chains, as well as receiving bridge messages to either redeem tokens it holds as collateral (i.e. unlock), or route them to other spoke chains (i.e. "multi-hop"). In the case of a multi-hop transfer, the `TokenHub` already has the collateral locked from when the tokens were originally bridged to the first spoke chain, so it simply updates the accounting of the transferred balances to each respective spoke. Spoke contracts must first be registered with a `TokenHub` instance before the hub contract will allow for sending tokens to them. This is to prevent tokens from being bridged to invalid spoke addresses. Anyone is able to deploy and register spoke contracts, which may have been modified from this repository. It is the responsibility of the users of the hub contract to independently evaluate each spoke for its security and correctness.
```